### PR TITLE
ENH: better construction of dict arrays and distance matrices

### PR DESCRIPTION
**Pull request by GavinHuttley**

### CODE REVIEW

File: src/cogent3/core/new_sequence.py [hunk 1]
```diff
@@ -2596,42 +2636,48 @@ def copy(self, sliced: bool = False):
 
 
 @singledispatch
-def _coerce_to_seqview(data, seqid, alphabet) -> SeqViewABC:
+def _coerce_to_seqview(data, seqid, alphabet, offset) -> SeqViewABC:
```
Comment (KatherineCaley) [inline src/cogent3/core/new_sequence.py:2639]: in the old sequence the new argument is called `annotation_offset`, otherwise the changes are the same

**Review by KatherineCaley — APPROVED** [review]

File: src/cogent3/__init__.py [hunk 2]
```diff
@@ -323,7 +332,7 @@ def load_seq(
     name, seq = data[0]
     name = label_to_name(name) if label_to_name else name
 
-    result = make_seq(seq, name, moltype=moltype, new_type=new_type)
+    result = make_seq(seq, name, moltype=moltype, new_type=new_type, **kw)
```
Comment (KatherineCaley) [inline src/cogent3/__init__.py:335]: should you be passing `annotation_offset` to `make_seq` here?

File: src/cogent3/core/new_sequence.py (hunk 1 above)
Comment (GavinHuttley) [reply src/cogent3/core/new_sequence.py:2639]: Given the old module is destined for removal, I'm going to let that inconsistency fade gracefully into oblivion!

File: src/cogent3/__init__.py (hunk 2 above)
Comment (GavinHuttley) [reply src/cogent3/__init__.py]: Good catch!

**Pull request merged**